### ISSUE TITLE
Lay out CJK text vertically in point-placed labels

### ISSUE THREAD

**Issue by 1ec5** [mapbox/mapbox-gl-js#3470]
As the next step after #1246, we should allow point-placed labels containing Chinese, Japanese, or Korean text to be laid out as vertical text **only if** a horizontal layout would collide with another feature. Vertical layout means that characters progress from top to bottom and lines progress from right to left, while the glyphs themselves remain unrotated. As discussed in https://github.com/mapbox/mapbox-gl-js/pull/3438#discussion_r84585841, Mongolian is written top to bottom, left to right, with glyph rotation; however, that script would be a much lower priority than CJK.

The primary use case for this feature is to better accommodate modern CJK text that can be written vertically if needed but is usually written horizontally. For antique-looking CJKV maps, it would be desirable to support a `writing-mode` layout property that forces vertical text even when horizontal text could be accommodated.

/cc @lucaswoj @xrwang @nickidlugash @andreasviglakis

**Comment by nickidlugash** [comment]
@1ec5 I don't know that it makes sense to support this before we support alternative horizontal placements of point labels to minimize collisions. I think this will make our maps harder to read, and most Chinese maps at least don't do this.

**Comment by 1ec5** [comment]
I won’t speak to priorities; CJK text is an area of continuous improvement rather than a project we can call complete. I want to reiterate this clause in the original description:

> when a horizontal layout would collide with another feature

That is, I fully agree that this feature would depend on alternative label placements. Furthermore, this feature is indeed common in Japanese maps, as seen for instance in https://github.com/mapbox/mapbox-gl-native/issues/1682#issuecomment-108606199.

**Comment by nickidlugash** [comment]
>  I fully agree that this feature would depend on alternative label placements. 

👍  Ok great, my concern was regarding push something live for vertical placements before alternative horizontal placements went live.

**Comment by asheemmamoowala** [comment]
Moving in #8399

**Comment by asheemmamoowala** [comment]
Closed in #8399
